src/client/mod.rs

```diff
@@ -152,19 +151,20 @@ impl APIClient {
     pub async fn request_events<T>(
         &self,
         request: http::Request<Vec<u8>>,
-    ) -> Result<impl Stream<Item = Result<T>>>
+    ) -> Result<impl TryStream<Item = Result<T>>>
     where
         T: DeserializeOwned,
     {
         let res: reqwest::Response = self.send(request).await?;
         trace!("{} {}", res.status().as_str(), res.url());
 
-        // Now use `unfold` to convert the chunked responses into a Stream
+        // Now unfold the chunked responses into a Stream
         // We first construct a Stream of Vec<Result<T>> as we potentially might need to
         // yield multiple objects per loop, then we flatten it to the Stream<Result<T>> as expected.
         // Any reqwest errors will terminate this stream early.
-        let stream = futures::stream::unfold((res, Vec::new()), |(mut resp, mut buff)| {
+        let stream = futures::stream::try_unfold((res, Vec::new()), |(mut resp, _buff)| {
             async {
+                let mut buff = _buff;
```

Comment:
for some reason, the try_unfold was more iffy when it came to reusing returning things referencing buff